src/with-theme.js

```diff
@@ -0,0 +1,46 @@
+import React, {Component} from 'react'
+
+import {CHANNEL} from './theme-provider'
+import {PropTypes} from './react-compat'
+
+export default function withTheme(ComponentToTheme) {
+  class ThemedComponent extends Component {
+    state = {theme: {}}
+    setTheme = theme => this.setState({theme})
+
+    componentWillMount() {
+      if (!this.context[CHANNEL]) {
+        /* istanbul ignore if */
+        if (process.env.NODE_ENV === 'production') {
```

Comment:
This should be: `if (process.env.NODE_ENV !== 'production') {`

We don't want to warn in production :)